libs/ui/src/lib/carousel/carousel.component.ts

```diff
@@ -50,13 +72,31 @@ export class CarouselComponent implements AfterViewInit, AfterContentInit, OnDes
   private currentPosition: number;
 
   @ViewChild(CdkScrollable) scrollable: CdkScrollable;
-  @ViewChild('container') container: ElementRef<HTMLDivElement>;
   @ContentChildren(CarouselItemDirective) items: QueryList<CarouselItemDirective>;
 
-  constructor(private cdr: ChangeDetectorRef, private ngZone: NgZone) { }
+  @Input() set columns(amount: Partial<Columns> | string) {
+    if (!amount) return;
+
+    if (typeof amount === 'object') {
+      for (const [size, values] of Object.entries(sizes)) {
+        const index = Object.keys(amount).find(key => values.includes(key));
+        if (index) {
+          this.el.nativeElement.style.setProperty(`--${size}-columns`, amount[index]);
+        }
+      }
```

Comment:
with the new system : 
```suggestion
      for (const [size, value] of Object.entries(amount)) {
        sizes[size]?.forEach(label => this.el.nativeElement.style.setProperty(`--${label}-columns`, value));
      }
```